### PR TITLE
Make docker_swarm_service option publish.published_port optional

### LINKED ISSUE #99: docker_swarm_service publish.published_port should not be mandatory.
##### SUMMARY

`docker_swarm_service` `publish.published_port` option should not be mandatory. It is not mandatory option in docker swarm. If not provided a random high port is assigned. Example this works fine:

```
docker service create --name my_web --replicas 3 --publish target=80 nginx
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`docker_swarm_service` module

##### ANSIBLE VERSION

```paste below
ansible 2.9.6

```

### PR DESCRIPTION
**Pull request by sgpinkus**
##### SUMMARY
Change docker_swarm_service option publish.published_ported to optional. If not specified random high port is assigned by docker. Fixes #99.

##### ISSUE TYPE

- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
docker_swarm_service

##### ADDITIONAL INFORMATION
This now works A free high port above 30000 will be assigned by docker:

```
- hosts: swarm-master
  tasks:
    - name: add swarm service
      community.docker.docker_swarm_service:
        state: present
        name: my-test-service
        replicas: 1
        image: nginx
        publish:
          - target_port: 80
```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
1. This needs a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).
2. The remaining sanity error (`no-log-required`) will be fixed by #102, just ignore it.

CC maintainers @dariko @jwitko @hannseman @WojciechowskiPiotr

**Comment by WojciechowskiPiotr** [comment]
Happy to merge it when the changelog is updated.

**Comment by sgpinkus** [comment]
Added, and squashed into one commit. Thanks!

**Comment by WojciechowskiPiotr** [comment]
@felixfontein can you take a look at why tests are still failing? #102 has been merged.

File: changelogs/fragments/101-make-service-published-port-optional.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - docker_swarm_service - change publish.published_port option from mandatory
```
Comment (felixfontein) [inline changelogs/fragments/101-make-service-published-port-optional.yaml:2]:
```suggestion
  - docker_swarm_service - change ``publish.published_port`` option from mandatory
```

**Comment by felixfontein** [comment]
@WojciechowskiPiotr they failed because collection installation from galaxy failed (after three trials). That seems to happen a lot more since yesterday or so (https://github.com/ansible/galaxy/issues/2429). Once the suggestion is merged tests will run another time anyway, and if some fail again I'll kick them until they pass :)

**Comment by felixfontein** [comment]
> Happy to merge it when the changelog is updated.

Everything looks good to me (but then I'm not that familiar with Docker swarm), so feel free to merge.

I'll create a new release once this is merged to get the security fix out.

**Comment by felixfontein** [comment]
@sgpinkus thanks for fixing this!
@WojciechowskiPiotr thanks for reviewing and merging!

**Pull request merged**